resolver/resolver.go

```diff
@@ -100,8 +108,17 @@ func (res *Resolver) records() *records.RecordGenerator {
 func (res *Resolver) LaunchDNS() <-chan error {
 	// Handers for Mesos requests
 	dns.HandleFunc(res.config.Domain+".", panicRecover(res.HandleMesos))
-	// Handler for nonMesos requests
-	dns.HandleFunc(".", panicRecover(res.HandleNonMesos))
+	// Handlers for nonMesos requests
+	for zone, fwd := range res.fwds {
```

Comment:
We could sort them by length descending to ensure that the most specific matcher wins. I'm fine with validation that there are no overlaps, also.